var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -157,6 +157,13 @@ class Dealii(CMakePackage, CudaPackage):
     depends_on('trilinos@master+amesos+aztec+epetra+ifpack+ml+muelu+rol+sacado+teuchos~amesos2~ifpack2~intrepid2~kokkos~tpetra~zoltan2',       when='+trilinos+mpi~int64+cuda')
     depends_on('trilinos@master+amesos+aztec+epetra+ifpack+ml+muelu+rol+sacado+teuchos~hypre~amesos2~ifpack2~intrepid2~kokkos~tpetra~zoltan2', when='+trilinos+mpi+int64+cuda')
 
+    # Both Trilinos and SymEngine bundle the Teuchos RCP library.
+    # This leads to conflicts between macros defined in the included
+    # headers when they are not compiled in the same mode.
+    # See https://github.com/symengine/symengine/issues/1516
+    depends_on("symengine~debug", when="^symengine ^trilinos~debug")
```

Comment:
do you need `^symengine` here? I think `when="^trilinos~debug")` should be enough.

Reply:
It looks like one does need this. Without it, I get the following error (note concretising without `SymEngine`):
```
$ ./spack spec dealii@9.1~petsc~slepc+trilinos~symengine
Input spec
--------------------------------
dealii@9.1~petsc~slepc~symengine+trilinos

Concretized
--------------------------------
==> Error: multiple values are not allowed for variant "build_type"
```

I understand that this specified that if one is building with SymEngine and Trilinos enabled, then SymEngine must be built with the specified settings. With just `depends_on("symengine~debug", when="^trilinos~debug")`, I believe it is stating a dependency of SymEngine if building with Trilinos enabled, whether or not SymEngine is enabled.